src/extension/components/ekubo_oracle.cairo

```diff
@@ -61,13 +94,22 @@ mod ekubo_oracle_component {
         /// * `price` - current price of the asset
         /// * `valid` - always `true`
         fn price(self: @ComponentState<TContractState>, pool_id: felt252, asset: ContractAddress) -> (u256, bool) {
-            let EkuboOracleConfig { oracle_pool, quote_token, period } = self
+            let EkuboOracleConfig { quote_token, quote_token_decimals, period } = self
                 .ekubo_oracle_configs
                 .read((pool_id, asset));
-            let oracle = IEkuboOracleDispatcher { contract_address: oracle_pool };
-            let price = oracle.get_price_x128_over_last(asset, quote_token, period);
+            let oracle = IEkuboOracleDispatcher { contract_address: self.oracle_address.read() };
+            let mut price = oracle.get_price_x128_over_last(asset, quote_token, period);
+
+            let asset_decimals: u8 = ERC20ABIDispatcher { contract_address: asset }.decimals();
+            // Adjust the scale based on the difference in precision between the base asset 
+            // and the quote asset
+            let adjusted_scale = if quote_token_decimals <= asset_decimals {
+                SCALE * pow_10((asset_decimals - quote_token_decimals).into())
+            } else {
+                SCALE / pow_10((quote_token_decimals - asset_decimals).into())
+            };
 
-            let price = price * SCALE / TWO_128;
+            let price = price * adjusted_scale / TWO_128;
```

Comment:
I modified this section to convert the price based on the difference in precision between the two assets. For example, a quote of ETH/USDC needs to be scaled up by their precision difference, whereas a quote of USDC/ETH needs to be scaled down by their precision difference.